test/js/specs/unit/collections/AccessPolicy.spec.js

```diff
@@ -0,0 +1,1519 @@
+        });
+      });
+
+      it("parse consistently sets dataONEObject on every rule to the collection's current reference", function() {
+        const ap = new AccessPolicy();
+        ap.dataONEObject = d1oA;
+
+        const xml = createAccessPolicyElement([
+          { subject: "s1", read: true },
+          { subject: "s2", write: true }
+        ]);
+        ap.parse(xml);
+
+        expect(ap.at(0).get("dataONEObject")).to.equal(d1oA);
+        expect(ap.at(1).get("dataONEObject")).to.equal(d1oA);
+
+        // Change the collection's dataONEObject and parse another XML; new values should reflect the new reference.
+        ap.dataONEObject = d1oB;
+        const xml2 = createAccessPolicyElement([{ subject: "s3", changePermission: true }]);
+        ap.parse(xml2);
+
+        expect(ap).to.have.lengthOf(1);
+        expect(ap.at(0).get("subject")).to.equal("s3");
+        expect(ap.at(0).get("dataONEObject")).to.equal(d1oB);
+      });
+    });
+
+    describe("preservation of rules during edits", function() {
+      let obj, policy;
+
+      beforeEach(function() {
+        // Create a DataONEObject for these tests
+        obj = new DataONEObject({ id: "test-preservation-obj" });
+        policy = obj.createAccessPolicy();
+
+        // Set up the specific state modifications needed for these tests
+        policy.add([
+          makeRule("uid:alice", ["read", "write"]),
+          makeRule("uid:bob", ["read"]),
+          makeRule("uid:carol", ["read", "changePermission"]),
+          makeRule("public", ["read"])
+        ]);
+      });
+
+      afterEach(function() {
+        // Clean up after each test in this block
+        if (policy) {
+          policy.reset();
+        }
+        obj = null;
+        policy = null;
+      });
+
+      it("does not remove other rules when a single AccessRule is updated in place", function() {
+        const initialCount = policy.length;
+        expect(initialCount).to.equal(4);
+
+        // Update bob's permissions in place
+        const bobRule = policy.findWhere({ subject: "uid:bob" });
+        bobRule.set("write", true);
+
+        expect(policy.length).to.equal(4);
+        expect(bobRule.get("read")).to.be.true;
+        expect(bobRule.get("write")).to.be.true;
+      });
+
+      it.skip("EXPECTED FAILURE: set() with partial list should NOT remove rules not included", function() {
+        const editedBob = new AccessRule({
+          subject: "uid:bob",
+          read: true,
+          write: true,
+          dataONEObject: obj
+        });
+
+        policy.set([editedBob]);
+
+        // This should fail - we expect ALL rules to be preserved
+        expect(policy.length).to.equal(4, "All rules should be preserved");
+        expect(policy.findWhere({ subject: "uid:alice" })).to.exist;
+        expect(policy.findWhere({ subject: "uid:carol" })).to.exist;
+        expect(policy.findWhere({ subject: "public" })).to.exist;
+      });
```

Comment:
Actually, this is how we expect `collection.set` to work, see the [backbone.js docs](https://backbonejs.org/#Collection-set):

> ... and if the collection contains any models that aren't present in the list, they'll be removed

We can pass the options `{ remove: false }` to `set` to keep other models, but then we will actually have 5 rules, not 4 as your test asserts. I believe backbone will only merge the two models if the model IDs are the same (pretty sure, but verify)